sql/moz-fx-data-shared-prod/bigeye_derived/collection_v2_service_v1/query.py

```diff
@@ -0,0 +1,105 @@
+"""Pull from BigEye API from Collection V2 Service and upload to Big Query."""
+
+import logging
+import os
+import re
+from argparse import ArgumentParser
+from typing import Any, Dict
+
+import pandas as pd
+import requests
+from google.cloud import bigquery
+
+from bigquery_etl.config import ConfigLoader
+
+BIGEYE_API_KEY = os.environ["BIGEYE_API_KEY"]
+WORKSPACE_IDS = ConfigLoader.get("monitoring", "bigeye_workspace_ids")
+API_URL = "https://app.bigeye.com/api/v2/collections/info?workspaceId="
+
+
+def make_api_request(workspace_id: int) -> Dict[str, Any]:
+    """Make API request to Bigeye for a specific workspace."""
+    headers = {
+        "accept": "application/json",
+        "authorization": BIGEYE_API_KEY,
+    }
+
+    response = requests.get(API_URL + str(workspace_id), headers=headers)
+    response.raise_for_status()
+    return response.json()
+
+
+def process_response(response_data: Dict[str, Any]) -> pd.DataFrame:
+    """Process API response into a DataFrame."""
+    df = pd.json_normalize(
+        response_data["collectionInfos"],
+        record_path=None,
+        meta=["collectionMetricStatus"],
+        meta_prefix=None,
+        record_prefix="",
+        errors="raise",
+        sep="_",
+        max_level=None,
+    )
+
+    column_list = df.columns.tolist()
+    for column_name in column_list:
+        name = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", column_name)
+        snake_case_name = re.sub("([a-z0-9])([A-Z])", r"\1_\2", name).lower()
+        df = df.rename(columns={f"{column_name}": f"{snake_case_name}"})
+
+    df["refreshed_at"] = pd.Timestamp.now(tz="UTC")
+
+    return df
+
+
+def get_bigeye_data() -> pd.DataFrame:
+    """Fetch data from Bigeye API for all workspaces and combine into a single DataFrame."""
+    all_data = []
+
+    for workspace_id in WORKSPACE_IDS:
+        try:
+            response_data = make_api_request(workspace_id)
+            df = process_response(response_data)
+            all_data.append(df)
+            logging.info(f"Successfully processed data for workspace {workspace_id}")
+        except Exception as e:
+            logging.info(f"Error processing workspace {workspace_id}: {str(e)}")
+            continue
```

Comment:
```suggestion
```

It looks like this is already the end of the loop? We probably don't need the continue here.